lib/sms/sms_submit.c

```diff
@@ -0,0 +1,360 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <logging/log.h>
+#include <stdio.h>
+#include <zephyr.h>
+#include <errno.h>
+#include <modem/at_cmd.h>
+#include <modem/sms.h>
+
+#include "string_conversion.h"
+
+
+LOG_MODULE_DECLARE(sms, CONFIG_SMS_LOG_LEVEL);
+
+/** @brief User Data Header size in octets/bytes. */
+#define SMS_UDH_CONCAT_SIZE_OCTETS 6
+/** @brief User Data Header size in septets. */
+#define SMS_UDH_CONCAT_SIZE_SEPTETS 7
+/** @brief Maximum length of the response for AT commands. */
+#define SMS_AT_RESPONSE_MAX_LEN 256
```

Comment:
We allocate this buffer on the stack in multiple places, see my other comments.

Reply:
This one is likely easier than some others. We clearly save stack by putting one instance of a buffer of this size as static and use it in various places.

Reply:
Not used anymore. Constant removed.